XPlanning/src/examples/mobilerobot/models/RobotLocationFormula.java

```diff
@@ -28,9 +28,11 @@
 
 	private StateVarDefinition<Location> mrLocDef;
 	private EffectClass mEffectClass; // of rLoc
+	private StateVarDefinition<HeadlampState> mrHeadlampDef;
```

Comment:
Movement probability now depends on the headlamp state, so change this file to add headlamp as a discriminant class and to use it to calculate transition probability.